src/validation/validation.cpp

```diff
@@ -2748,7 +2748,10 @@ void UpdateTip(CBlockIndex *pindexNew)
     else
     {
         maxScriptOps = MAX_OPS_PER_SCRIPT;
-        excessiveBlockSize = DEFAULT_EXCESSIVE_BLOCK_SIZE;
+        if (chainParams.NetworkIDString() != "regtest")
+        {
+            excessiveBlockSize = DEFAULT_EXCESSIVE_BLOCK_SIZE;
+        }
```

Comment:
I disagree with this...problem is if users on mainnet want to change their Excessive block size manually it won't stick after the next block is mined...I think there's a logic error somewhere in your new code changes.

Reply:
Also this does not do what the commit message says it does

Reply:
Agreed I should have split the `if` condition in 2, something like: 

```c++
    if (AreWeOnSVChain())
    {
        if (IsSv2018Activated(chainParams.GetConsensus(), chainActive.Tip()))
        {
            maxScriptOps = SV_MAX_OPS_PER_SCRIPT;
            excessiveBlockSize = SV_EXCESSIVE_BLOCK_SIZE;
        }
        else // if blockchain reorg we may need to back it out
        {
            maxScriptOps = MAX_OPS_PER_SCRIPT;
            excessiveBlockSize = DEFAULT_EXCESSIVE_BLOCK_SIZE;
        }
    }
```

current `else` branch was supposed to be execute only in the BSV case, on the contrary the current code execute the `else` branch also BCH branch which is a logic bug. 

fixing.

Reply:
fixed ea81d7a3c7